### PR TITLE
Prac 08 feedback

### PR DESCRIPTION
**Pull request by DamonLindsay**
Hi @jasonkennedy01, When you have a moment could you leave some feedback for me, would love to hear any improvements you have to share 🌮  😋

### CODE REVIEW

File: prac_08/convert_miles_km.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def build(self):
+        """Construct the app."""
+        self.title = "Convert Miles to Kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        self.message = "0.0"
+        return self.root
+
+    def handle_update(self):
+        """Handle text input changes and update the displayed result."""
+        miles = self.root.ids.user_input.text
```
Comment (jasonkennedy01) [inline prac_08/convert_miles_km.py:26]: Could you have included the value in the method call within the kv file, like in the squaring program?

File: prac_08/convert_miles_km.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def handle_increment(self, increment_value):
+        """Increment or decrement the miles based on the given value."""
+        current_miles = self.root.ids.user_input.text
+        try:
+            if not current_miles:
+                updated_miles = increment_value
+            else:
+                miles = float(current_miles)
+                updated_miles = miles + increment_value
+            self.root.ids.user_input.text = str(updated_miles)
```
Comment (jasonkennedy01) [inline prac_08/convert_miles_km.py:42]: I believe this is handled by the ValueError exception.

File: prac_08/convert_miles_km.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+"""
+Estimated: 90 minutes
+Actual: 70 minutes
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.properties import StringProperty
+
+MILES_TO_KILOMETRES_CONVERSION_RATE = 1.60934  # 1 kilometer
```
Comment (jasonkennedy01) [inline prac_08/convert_miles_km.py:10]: Nice constant name and comment.

File: prac_08/dynamic_labels.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class DynamicLabelsApp(App):
+    """Dynamic Labels App to dynamically create labels based on information given."""
+
+    def __init__(self, **kwargs):
+        """Initialize the Dynamic Labels App object."""
+        super().__init__(**kwargs)
+        self.name_to_favourite_colour = {"Damon Lindsay": "Purple", "Liam Lindsay": "Green", "Jordan Lindsay": "Pink",
+                                         "Camryn Lindsay": "Blue"}
```
Comment (jasonkennedy01) [inline prac_08/dynamic_labels.py:18]: I think the question asked for a list of names not a dictionary.

**Review by jasonkennedy01 — COMMENTED** [review]
Well done @DamonLindsay 😄 , very good docstrings and coding principles. I'm not sure why prac_06 and prac_07 are in this PR however.  Keep up the good work 👑.

File: prac_08/dynamic_labels.py (hunk 4 above)
Comment (DamonLindsay) [reply prac_08/dynamic_labels.py:18]: Woops!  Not sure where I got the dictionary from 😭  - good find!

**Comment by DamonLindsay** [comment]
> Well done @DamonLindsay 😄 , very good docstrings and coding principles. I'm not sure why prac_06 and prac_07 are in this PR however. Keep up the good work 👑.

They definitely werent meant to be, I think I made some mistakes and merged the other branches into this one instead of master... I'm trying 😿

File: prac_08/convert_miles_km.py (hunk 2 above)
Comment (DamonLindsay) [reply prac_08/convert_miles_km.py:42]: I'd have to double check, but I believe the 
```
            if not current_miles:
                updated_miles = increment_value              
```
handles the specific scenario when user input is empty (e.g. no value was entered).  If no value was entered then it sets the _updated_miles_ to the _increment_value_.

The _ValueError_ catches situations where the entered value cannot be converted to a float.

So I think they are both okay as they handle different issues...  But I guess when Lindsay reviews he can provide his 2cents 😄

File: prac_08/convert_miles_km.py (hunk 1 above)
Comment (DamonLindsay) [reply prac_08/convert_miles_km.py:26]: Yeah this was a good catch, thank you!

**Pull request merged**